appinventor/components-ios/src/Trendline.swift

```diff
@@ -0,0 +1,592 @@
+//
+//  Trendline.swift
+//  AIComponentKit
+//
+//  Created by David Kim on 3/19/24.
+//  Copyright © 2024 Massachusetts Institute of Technology. All rights reserved.
+//
+
+import Foundation
+import DGCharts
+import Network
+import UIKit
+
+@objc public class BestFitModel: NSObject, OptionList {
+  @objc public static let Linear = BestFitModel(0)
+  @objc public static let Quadratic = BestFitModel(1)
+  // @objc public static let Cubic = BestFitModel("Cubic")
+  @objc public static let Exponential = BestFitModel(2)
+  @objc public static let Logarithmic = BestFitModel(3)
+
+  private static let LOOKUP: [Int32: BestFitModel] = [
+    0: .Linear,
+    1: .Quadratic,
+    // "Cubic": .Cubic,
+    2: .Exponential,
+    3: .Logarithmic
+  ]
+
+  let value: Int32
+
+  @objc private init(_ value: Int32) {
+    self.value = value
+  }
+
+  @objc public class func fromUnderlyingValue(_ value: Int32) -> BestFitModel? {
+    return LOOKUP[value]
+  }
+
+  @objc public func toUnderlyingValue() -> AnyObject {
+    return value as AnyObject
+  }
+}
+
+@objc public class StrokeStyle: NSObject {
+  @objc public static let Solid = StrokeStyle(1)
+  @objc public static let Dashed = StrokeStyle(2)
+  @objc public static let Dotted = StrokeStyle(3)
+
+  private static let lookup: [Int: StrokeStyle] = [
+    1: Solid,
+    2: Dashed,
+    3: Dotted
+  ]
+
+  let value: Int
+
+  @objc private init(_ value: Int) {
+    self.value = value
+  }
+
+  @objc public class func fromUnderlyingValue(_ value: Int) -> StrokeStyle? {
+    return lookup[value]
+  }
+
+  //TODO: how can we resolve this?
+  //  @objc public class func fromUnderlyingValue(_ value: String) -> StrokeStyle? {
+  //    if let intValue = Int(value) {
+  //      return fromUnderlyingValue(intValue)
+  //    }
+  //    return nil
+  //  }
```

Comment:
I think for this we will need to have the function take `AnyObject` and then handle the `Int` vs `String` issue internally since Objective-C doesn't have method overloading.